hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BaseSparkCommitActionExecutor.java

```diff
@@ -103,6 +104,9 @@ public BaseSparkCommitActionExecutor(HoodieEngineContext context,
     if (isWorkloadProfileNeeded()) {
       profile = new WorkloadProfile(buildProfile(inputRecordsRDD));
       LOG.info("Workload profile :" + profile);
+      // apply clustering update strategy.
```

Comment:
I think we need to think more about the algorithm here.  For example, 
1. f1, f2, f3 are file groups in partition.
2. Assume there is pending clustering on all file groups f1, f2, f3.
3. f3 is a small file. So we buildProfile would assign inserts to f3.
4. applying update strategy will throw error because f3 is included.

Instead, we may want to change buildProfile to exclude file groups that are in pending clustering. So, new file f4 would be created in step#3 and ingestion can continue. This way inserts can continue without errors.

Reply:
this is a good case , i will do it in build profile